site/src/pages/OrganizationSettingsPage/Sidebar.tsx

```diff
@@ -0,0 +1,179 @@
+import { cx } from "@emotion/css";
+import type { FC, ReactNode } from "react";
+import { Link, NavLink } from "react-router-dom";
+import type { Organization } from "api/typesGenerated";
+import { Sidebar as BaseSidebar } from "components/Sidebar/Sidebar";
+import { Stack } from "components/Stack/Stack";
+import { UserAvatar } from "components/UserAvatar/UserAvatar";
+import { type ClassName, useClassName } from "hooks/useClassName";
+import { useOrganizationSettings } from "./OrganizationSettingsLayout";
+
+export const Sidebar: FC = () => {
+  const { currentOrganizationId, organizations } = useOrganizationSettings();
+
+  // maybe do something nice to scroll to the active org
+
+  return (
+    <BaseSidebar>
+      {organizations.map((organization) => (
+        <OrganizationBloob
+          key={organization.id}
+          organization={organization}
+          active={organization.id === currentOrganizationId}
+        />
+      ))}
+    </BaseSidebar>
+  );
+};
+
+interface BloobProps {
+  organization: Organization;
+  active: boolean;
+}
+
+function urlForSubpage(organizationName: string, subpage: string = ""): string {
+  return `/organizations/${organizationName}/${subpage}`;
+}
+
+export const OrganizationBloob: FC<BloobProps> = ({ organization, active }) => {
+  return (
+    <>
+      <SidebarNavItem
+        active={active}
+        href={urlForSubpage(organization.name)}
+        icon={
+          <UserAvatar
+            key={organization.id}
+            size="sm"
+            username={organization.display_name}
+            avatarURL={organization.icon}
+          />
+        }
+      >
+        {organization.display_name}
+      </SidebarNavItem>
+      {active && (
+        <Stack spacing={0.5} css={{ marginBottom: 8, marginTop: 8 }}>
+          <SidebarNavSubItem href={urlForSubpage(organization.name)}>
+            Organization settings
+          </SidebarNavSubItem>
+          <SidebarNavSubItem
+            href={urlForSubpage(organization.name, "external-auth")}
+          >
+            External authentication
+          </SidebarNavSubItem>
+          <SidebarNavSubItem href={urlForSubpage(organization.name, "members")}>
+            Members
+          </SidebarNavSubItem>
+          <SidebarNavSubItem href={urlForSubpage(organization.name, "groups")}>
+            Groups
+          </SidebarNavSubItem>
+          <SidebarNavSubItem href={urlForSubpage(organization.name, "metrics")}>
+            Metrics
+          </SidebarNavSubItem>
+          <SidebarNavSubItem
+            href={urlForSubpage(organization.name, "auditing")}
+          >
+            Auditing
+          </SidebarNavSubItem>
+        </Stack>
+      )}
+    </>
+  );
+};
+
+interface SidebarNavItemProps {
+  active?: boolean;
+  children?: ReactNode;
+  icon: ReactNode;
+  href: string;
+}
+
+export const SidebarNavItem: FC<SidebarNavItemProps> = ({
```

Comment:
I think we have a set of common components to build a sidebar. If not, we probably should make them to keep it consistent.

Reply:
we do, but the styling of the sidebar in this view needs to be quite different from our existing sidebars. we can see if there's a nice way to unify them later. I tried using them initially, but it's a lot easier to iterate this way for now.